src/NuGet.Core/NuGet.ProjectModel/JsonUtility.cs

```diff
@@ -55,7 +55,7 @@ internal static JProperty WritePackageDependency(PackageDependency item)
         {
             return new JProperty(
                 item.Id,
-                WriteString(item.VersionRange?.ToLegacyShortString()));
```

Comment:
Instead of change this WritePackageDependency, I'd just change what the PackagesLockFileFormat uses. 

Basically right now, PackagesLockFileFormat and LockFileFormat both use the same method, but we probably don't want to change the assets file.

Reply:
Shouldn't we use ToString instead here? It's the functional version that matters.